### PR TITLE
7-post reviews

### PR DESCRIPTION
**Pull request by reawilliams20**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -164,4 +159,66 @@ describe('/api/reviews/:review_id', () => {
...
+                  comment_id: 7,
+                  body: "some text...",
+                  review_id: review_id,
+                  "author": "mallionaire",
+                  "votes": 0,
+                  created_at: expect.any(String)
+                
+              })
+            )
+        })
```
Comment (RichardNeat) [inline __tests__/app.test.js:187]: Could we add a test for 201 - ignoring unnecessary properties? If extra keys and values get sent over can we prove that they will be ignored correctly?

File: __tests__/app.test.js [hunk 2]
```diff
@@ -164,4 +159,66 @@ describe('/api/reviews/:review_id', () => {
...
+        .post(`/api/reviews/${review_id}/comments`)
+        .send({"body": "test body",
+        "other-property": "mallionaire"})
+        .expect(400)
+        .send((response) => {
+            const msg = response.body.msg
+            expect(msg).toBe('bad request')
+        })
+    });
+})
```
Comment (RichardNeat) [inline __tests__/app.test.js:272]: I think we need to consider a couple of other sad path cases, how about what can go wrong with the ID? Valid but non existent and invalid ID?

**Review by RichardNeat — CHANGES_REQUESTED** [review]
Great work, just a couple of extra considerations for testing below please then resubmit for approval :)

File: app.js [hunk 3]
```diff
@@ -1,19 +1,24 @@
 const express = require("express");
 const {getCategories} = require("./controllers/controllers.categories")
-const {getReviews, getReview, getReviewComments} = require("./controllers/controllers.reviews");
-const {handle404Routes, handle400Error, handle500Error} = require("./error-handling")
+const {getReviews, getReview, getReviewComments,postComment} = require("./controllers/controllers.reviews");
+const {handle404Routes, handle400Error, handle500Error, handle404Error} = require("./error-handling")
```
Comment (lewispricey) [inline app.js:4]: Like the way you've split the error handling out to a seperate file to keep the app.js as readable as possible 👍

File: controllers/controllers.reviews.js [hunk 4]
```diff
@@ -1,4 +1,4 @@
-const {selectReviews, selectReview, selectReviewComments} = require("../models/models.reviews");
+const {selectReviews, selectReview, selectReviewComments, insertComment} = require("../models/models.reviews");
```
Comment (lewispricey) [inline controllers/controllers.reviews.js:1]: Great consistent naming of your model functions!

File: controllers/controllers.reviews.js [hunk 5]
```diff
@@ -31,4 +31,15 @@ exports.getReviewComments = (req, res, next) => {
...
     })
-}
\ No newline at end of file
+}
+exports.postComment = (req, res, next) => {
+    const review_id = req.params.review_id
+    const newComment = req.body
+    insertComment(newComment, review_id)
+    .then((comment) => {
+        res.status(201).send({comment})
```
Comment (lewispricey) [inline controllers/controllers.reviews.js:40]: Nice choice of key to send the data back on.

File: models/models.reviews.js [hunk 6]
```diff
@@ -27,8 +27,19 @@ exports.selectReviewComments = (review_id) => {
...
\ No newline at end of file
+}
+
+exports.insertComment = (newComment, review_id) => {
+    const {body, username} = newComment;
+    return db.query(`
+    INSERT INTO comments
+    (body, author, review_id)
+    VALUES
+    ($1, $2, $3)
```
Comment (lewispricey) [inline models/models.reviews.js:42]: Nice use of parameterized query to prevent SQL injection attacks.

File: __tests__/app.test.js [hunk 7]
```diff
@@ -164,4 +159,114 @@ describe('/api/reviews/:review_id', () => {
...
+    test("201: created, ignoring extra properties", () => {
+        const review_id = 1;
+        return request(app)
+        .post(`/api/reviews/${review_id}/comments`)
+        .expect(201)
+        .send({
+            "body": "some text...",
+            "username": "mallionaire",
+            "review_id": 2,
+            "votes": 3
```
Comment (lewispricey) [inline __tests__/app.test.js:198]: Really good test, like the way you used votes as an extra property so we could assert this is added to the database as 0 in the test

**Review by lewispricey — APPROVED** [review]
Great work Andrea, really comprehensive testing so we can be sure the endpoint will only work when the user passes in everything required to a valid ID.

Good to merge 🚀

**Pull request merged**